cdap-ui/app/cdap/api/market.js

```diff
@@ -35,11 +42,27 @@ export const MyMarketApi = {
     `/packages/:packageName/:version/spec.json`,
     requestOptions
   ),
-  getCategoryIcon: (category) => {
-    return `${basepath}/categories/${category}/icon.png`;
+  getCategoryIcon: (category, marketHost) => {
+    if (basepaths.length === 0) {
+      return `${basepath}/categories/${category}/icon.png`;
+    }
+
+    if (marketHost) {
+      const verifiedMarketHost = getVerifiedMarketBaseUrl(marketHost);
+      return `${verifiedMarketHost}/categories/${category}/icon.png`;
+    }
+    return `${basepaths[0].url}/categories/${category}/icon.png`;
```

Comment:
There could also be a case where basepaths could be empty right? If the user doesn't provide multiple market urls the property would be empty and we need to fallback to `basepath`

Reply:
Also assuming the default market would be at `0` index might not be valid all the time. The user can always change the order.